src/Konami.tsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+
+export interface KonamiProps {
+  className?: string;
+  code?: number[];
+  disabled?: boolean;
+  resetDelay?: number;
+  action?: () => void;
+  timeout?: number;
+  onTimeout?: () => void;
+}
+
+export interface KonamiStates {
+  done: boolean;
+  input: number[];
+}
+
+class Timer {
+  t: number;
+  timerObj: number | null;
```

Comment:
In hindsight it would be nice for me to have named this `timerIntervalID` or something like that. This is not an object.

Reply:
done